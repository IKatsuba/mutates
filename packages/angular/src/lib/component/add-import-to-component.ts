import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addImportToComponent(
  classDeclaration: ClassDeclaration,
  importName: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'Component', 'imports', importName, {
    unique,
    forceToArray: true,
  });
}
