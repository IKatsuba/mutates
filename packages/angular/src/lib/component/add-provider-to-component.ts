import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addProviderToComponent(
  classDeclaration: ClassDeclaration,
  provider: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'Component', 'providers', provider, {
    unique,
    forceToArray: true,
  });
}
