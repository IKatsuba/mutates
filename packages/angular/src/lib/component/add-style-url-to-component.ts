import type { ClassDeclaration } from 'ts-morph';

import { pushToDecoratorArrayProperty } from '../helpers/push-to-decorator-array-property';

export function addStyleUrlToComponent(
  classDeclaration: ClassDeclaration,
  styleUrl: string,
  { unique = false }: { unique?: boolean } = {},
): void {
  pushToDecoratorArrayProperty(classDeclaration, 'Component', 'styleUrls', styleUrl, {
    unique,
  });
}
