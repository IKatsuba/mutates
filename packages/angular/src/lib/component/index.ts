export * from './add-import-to-component';
export * from './add-provider-to-component';
export * from './add-style-url-to-component';
