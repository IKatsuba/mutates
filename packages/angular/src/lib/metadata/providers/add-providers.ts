import { ClassDeclaration } from 'ts-morph';

import { pushToMetadataProperty } from '../push-to-metadata-property';

export function addProviders(ngEntity: ClassDeclaration, providers: string[]): void {
  pushToMetadataProperty(ngEntity, 'providers', providers);
}
