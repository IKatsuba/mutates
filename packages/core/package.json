{
  "name": "@mutates/core",
  "version": "0.0.0-development",
  "keywords": [
    "typescript",
    "ast",
    "mutations"
  ],
  "homepage": "https://github.com/IKatsuba/mutates",
  "repository": "https://github.com/IKatsuba/mutates",
  "license": "Apache-2.0",
  "contributors": [
    "Igor Katsuba <igor@katsuba.dev>"
  ],
  "type": "commonjs",
  "main": "./src/index.js",
  "typings": "./src/index.d.ts",
  "dependencies": {
    "minimatch": "9.0.3",
    "ts-morph": "^22.0.0",
    "tslib": "^2.3.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "authors": [
    "Igor Katsuba <igor@katsuba.dev>"
  ]
}
