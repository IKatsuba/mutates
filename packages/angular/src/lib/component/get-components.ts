import type { ClassDeclaration } from 'ts-morph';

import { getClasses, Pattern, Query, StructureType } from '@mutates/core';

export function getComponents(
  pattern: Pattern,
  query?: Query<Omit<StructureType<ClassDeclaration>, 'kind'>>,
): ClassDeclaration[] {
  return getClasses(pattern, query).filter(isComponent);
}

export function isComponent(declaration: ClassDeclaration): boolean {
  return !!declaration.getDecorator('Component');
}
