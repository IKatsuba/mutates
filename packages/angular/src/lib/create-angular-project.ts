import { Tree } from '@angular-devkit/schematics';
import { ProjectOptions } from 'ts-morph';

import { createProject } from '@mutates/core';

import { NgTreeFileSystem } from './ng-tree-file-system';

export function createAngularProject(tree: Tree, options?: Omit<ProjectOptions, 'fileSystem'>) {
  return createProject(new NgTreeFileSystem(tree), options);
}
