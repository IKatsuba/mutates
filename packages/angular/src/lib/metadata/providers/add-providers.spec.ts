import { Tree } from '@angular-devkit/schematics';

import { createSourceFile, readFileSync, saveProject } from '@mutates/core';

import { getComponents } from '../../component';
import { createAngularProject } from '../../create-angular-project';
import { createTestingTree } from '../../testing';
import { addProviders } from './add-providers';

describe('addProviders', () => {
  let host: Tree;
  beforeEach(() => {
    host = createTestingTree();

    createAngularProject(host);
  });

  it('should add providers to the app module', () => {
    createSourceFile(
      'src/main.ts',
      `
@Component({}) class AppComponent {}
`,
    );

    addProviders(getComponents('src/main.ts').at(0)!, ['AppService']);

    saveProject();

    expect(readFileSync('src/main.ts')).matchSnapshot();
  });
});
