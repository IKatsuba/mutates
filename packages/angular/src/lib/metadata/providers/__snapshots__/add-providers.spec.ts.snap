// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`addProviders > should add providers to the app module 1`] = `
"
@Component({
    providers: [AppService]
}) class AppComponent {}
"
`;
